server/src/main/java/keywhiz/service/daos/SecretSeriesDAO.java

```diff
@@ -321,59 +321,99 @@ public ImmutableList<SecretSeries> getSecretSeriesBatched(int idx, int num, bool
     return ImmutableList.copyOf(r);
   }
 
-  public void deleteSecretSeriesByName(String name) {
-    long now = OffsetDateTime.now().toEpochSecond();
+  public void hardDeleteSecretSeriesByName(String name) {
+    dslContext.transaction(configuration -> {
+      DSLContext dslContext = DSL.using(configuration);
+
+      SecretsRecord record = dslContext.select()
+          .from(SECRETS)
+          .where(SECRETS.NAME.eq(name))
+          .forUpdate()
+          .fetchOneInto(SECRETS);
+
+      hardDeleteSecretSeries(dslContext, record);
+    });
+  }
+
+  public void hardDeleteSecretSeriesById(Long id) {
+    dslContext.transaction(configuration -> {
+      DSLContext dslContext = DSL.using(configuration);
+
+      SecretsRecord record = dslContext.select()
+          .from(SECRETS)
+          .where(SECRETS.ID.eq(id))
+          .forUpdate()
+          .fetchOneInto(SECRETS);
+
+      hardDeleteSecretSeries(dslContext, record);
+    });
+  }
```

Comment:
In the existing code there was a lot of duplication between the by-id and by-name methods, even though only the lookup logic was (slightly) different and the deletion logic was identical. I went ahead and split out the deletion logic into a dedicated helper method to reduce duplication in the new methods and also applied this same pattern to the old methods. Net result: much cleaner code.